### PR TITLE
[UF-521] goTo(PlaceRequest place, HTMLElement addTo) doesn't cleanup if it's called twice.

### PR DESCRIPTION
**Pull request by ederign**

### CODE REVIEW

**Comment by ederign** [comment]
just some mins before the review, I'm changing the API.

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java [hunk 1]
```diff
@@ -200,21 +203,46 @@ public void goTo(final PlaceRequest place,
     @Override
     public void goTo(PlaceRequest place,
                      HasWidgets addTo) {
+
+        closeOpenPlacesAt(panelsOfThisHasWidgets(addTo));
         goToTargetPanel(place,
                         panelManager.addCustomPanel(addTo,
                                                     UnanchoredStaticWorkbenchPanelPresenter.class.getName()));
+        System.out.println();
```
Comment (porcelli) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java]: ????

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java [hunk 2]
```diff
@@ -30,6 +31,7 @@
 import javax.enterprise.inject.Produces;
 import javax.inject.Inject;
 
+import com.google.gwt.core.client.GWT;
```
Comment (porcelli) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java]: ??

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java [hunk 3]
```diff
@@ -208,13 +213,35 @@ public void goTo(PlaceRequest place,
...
 
+    private void closeOpenPlacesAt(Predicate<CustomPanelDefinition> filterPanels) {
+        customPanels.values().stream()
+                .filter(filterPanels)
+                .flatMap(p -> p.getParts().stream())
+                .forEach(part -> closePlace(part.getPlace()));
+    }
+
+    private Predicate<CustomPanelDefinition> panelsOfThisHTMLElement(HTMLElement addTo) {
+        Predicate<CustomPanelDefinition> filterHTMLElement = p -> p.getHtmlElementContainer().isPresent();
```
Comment (porcelli) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PlaceManagerImpl.java]: improve legibility, please.

**Comment by ederign** [comment]
jenkins retest this

**Pull request merged**